### PR TITLE
Fix issue #207: always set self.origin

### PR DESCRIPTION
**Pull request by cjerdonek**
This fixes issue #207.

It would be good if this could be committed before related PR's so the test can be present for the other PR's.

### CODE REVIEW

**Comment by cjerdonek** [comment]
Note that the test added here is useful independent of issue #207. For example, it checks some of what is documented in PR #199, namely the availability of the attributes it lists as being available from within `get_response_status()`.

File: websockets/test_client_server.py [hunk 1]
```diff
@@ -217,6 +218,35 @@ def test_protocol_custom_response_headers_list(self):
...
+        class SaveAttributesProtocol(WebSocketServerProtocol):
+            @asyncio.coroutine
+            def get_response_status(self, set_header):
+                attrs['origin'] = self.origin
+                attrs['path'] = self.path
+                attrs['raw_request_headers'] = self.raw_request_headers.copy()
+                attrs['request_headers'] = self.request_headers
+                status = yield from super().get_response_status(set_header)
+                return status
+        self.start_server(klass=SaveAttributesProtocol)
```
Comment (aaugustin) [inline websockets/test_client_server.py]: Some extra newlines would help readability.

File: websockets/test_client_server.py [hunk 2]
```diff
@@ -217,6 +218,35 @@ def test_protocol_custom_response_headers_list(self):
...
+        try:
+            self.start_client(path='foo/bar', origin='http://otherhost')
+            self.assertEqual(attrs['origin'], 'http://otherhost')
+            self.assertEqual(attrs['path'], '/foo/bar')
+            # To reduce test brittleness, only check one nontrivial aspect
+            # of the request headers.
+            self.assertTrue(('Origin', 'http://otherhost') in
```
Comment (aaugustin) [inline websockets/test_client_server.py]: `self.assertIn(('Origin', 'http://otherhost'), attrs['raw_request_headers'])`

File: websockets/test_client_server.py [hunk 3]
```diff
@@ -217,6 +218,35 @@ def test_protocol_custom_response_headers_list(self):
...
+                attrs['raw_request_headers'])
+            request_headers = attrs['request_headers']
+            self.assertTrue(isinstance(request_headers, email.message.Message))
```
Comment (aaugustin) [inline websockets/test_client_server.py]: `self.assertIsInstance(request_headers, http.client.HTTPMessage)`

(I changed the type a few minutes ago.)

**Comment by cjerdonek** [comment]
Thanks for the review! I can make those changes and commit.

Question: when it comes to committing to master, do you have a preference between (1) "squashing" and committing (with no merge commit), and (2) not squashing and merging the PR commits?

**Comment by cjerdonek** [comment]
Addressed review comments, squashed, and committed as e0915479ce582b0b69b551cc62ab2eb4e6cd39f9.

**Comment by aaugustin** [comment]
I prefer squashing when the history isn't clean (e.g. lots of small fixes until tests pass). If the PR has several logical commits, performing a rebase or a merge is fine. I don't care very much ;-)

**Pull request closed without merging**